pkgs/by-name/op/openscad-unstable/package.nix

```
@@ -124,9 +125,20 @@ clangStdenv.mkDerivation rec {
     # IPO
     "-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld"
     "-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON"
+
+    # The sources enable this for only apple. We turn it off globally anyway to stay
+    # consistent.
+    "-DUSE_QT6=OFF"
   ];
 
-  doCheck = true;
+  # tests rely on sysprof which is not available on darwin
+  doCheck = !stdenv.hostPlatform.isDarwin;
+
+  postInstall = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    mkdir $out/Applications
+    mv $out/bin/*.app $out/Applications
+    rmdir $out/bin || true
```

```suggestion
    rmdir $out/bin
```
Why should that fail?

---

In principle, due to other command-line oriented tools which have no extension — I think on Linux OpenSCAD has always been one executable, is it the same on macOS?

---

@7c6f434c can confirm that it doesn't work without the `|| true` ....

---

Should we drop the rmdir completely then?

---

Yep I get an error when either `|| true` or the whole `rm` line is removed:
```
error: patch failed: pkgs/by-name/op/openscad-unstable/package.nix:137
error: pkgs/by-name/op/openscad-unstable/package.nix: patch does not apply

```

---

This doesn't look like a problem normally seen during nix-build, which command gives _that_?

---

`nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`

---

This looks more like a need for rebase than a build failure.

---

Also, we seem to be talking about a single rebuild, so it should be enough just to build `openscad-unstable` directly.

---

Not sure why this command exists in the main package. From my testing, `bin` is an empty folder. I have removed the `|| true` part and tested it still builds `nix build -A openscad-unstable`